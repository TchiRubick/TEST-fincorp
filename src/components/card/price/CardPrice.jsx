import React from 'react';

import './CardPrice.scss';

import BaseButton from '../../buttons/base/BaseButton';
import tick from '../../../assets/img/tick.png';

const CardPrice = (props) => {
  const {
    color,
    title,
    price,
    contents,
    off,
    sticker,
  } = props;

  const renderContent = () => contents.map((c) => (
    <div className="content-price-card" key={c}>
      <img src={tick} alt="tick" />
      {c}
    </div>
  ));

  const renderOff = () => {
    if (!off) {
      return null;
    }

    return (
      <span className={`content-price-badge  __${color || ''}`}>
        {`save ${off} %`}
      </span>
    );
  };

  const renderSticker = () => {
    if (!sticker) {
      return null;
    }

    return <div className={`content-price-sticker  __${color || ''}`} />;
  };

  return (
    <div className={`comp-card-price __${color || ''}`}>
      {renderSticker()}
      <span className={`comp-card-price__title __${color || ''}`}>
        {title}
      </span>
      {renderOff()}
      <div className={`comp-card-price__price __${color || ''}`}>
        <span className="comp-card-price__price-unit">$</span>
        <span className="comp-card-price__price-value">{price}</span>
      </div>
      <div className="comp-card-price__content">
        {renderContent()}
      </div>
      <div className="comp-card-price__button">
        <BaseButton color={color}>
          Choose plan
        </BaseButton>
      </div>
    </div>
  );
};

export default CardPrice;
