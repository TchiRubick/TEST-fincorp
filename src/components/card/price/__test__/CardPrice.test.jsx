import React from 'react';
import ReactDom from 'react-dom';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import CardPrice from '../CardPrice';

const contents = ['test content'];

afterEach(cleanup);

it('renders good', () => {
  const div = document.createElement('div');
  ReactDom.render(<CardPrice color="green" title="test" price={1} contents={contents} off={10} sticker />, div);
});

it('as expected snapshot 1', () => {
  const tree = renderer.create(<CardPrice color="green" title="test" price={1} contents={contents} off={10} sticker />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('as expected snapshot 2', () => {
  const tree = renderer.create(<CardPrice color="green" title="test" price={1} contents={contents} />).toJSON();
  expect(tree).toMatchSnapshot();
});
