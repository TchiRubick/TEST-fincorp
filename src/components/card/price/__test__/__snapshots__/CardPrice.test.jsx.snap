// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as expected snapshot 1 1`] = `
<div
  className="comp-card-price __green"
>
  <div
    className="content-price-sticker  __green"
  />
  <span
    className="comp-card-price__title __green"
  >
    test
  </span>
  <span
    className="content-price-badge  __green"
  >
    save 10 %
  </span>
  <div
    className="comp-card-price__price __green"
  >
    <span
      className="comp-card-price__price-unit"
    >
      $
    </span>
    <span
      className="comp-card-price__price-value"
    >
      1
    </span>
  </div>
  <div
    className="comp-card-price__content"
  >
    <div
      className="content-price-card"
    >
      <img
        alt="tick"
        src="tick.png"
      />
      test content
    </div>
  </div>
  <div
    className="comp-card-price__button"
  >
    <button
      className="comp-button-base __green"
      type="button"
    >
      Choose plan
    </button>
  </div>
</div>
`;

exports[`as expected snapshot 2 1`] = `
<div
  className="comp-card-price __green"
>
  <span
    className="comp-card-price__title __green"
  >
    test
  </span>
  <div
    className="comp-card-price__price __green"
  >
    <span
      className="comp-card-price__price-unit"
    >
      $
    </span>
    <span
      className="comp-card-price__price-value"
    >
      1
    </span>
  </div>
  <div
    className="comp-card-price__content"
  >
    <div
      className="content-price-card"
    >
      <img
        alt="tick"
        src="tick.png"
      />
      test content
    </div>
  </div>
  <div
    className="comp-card-price__button"
  >
    <button
      className="comp-button-base __green"
      type="button"
    >
      Choose plan
    </button>
  </div>
</div>
`;
